raulin0/backup-juggler
# feat: size calculation funcionality

# Description
This pull request introduces new tests for the calculates_size function and adds the calculates_size function itself to the project. The tests ensure that the function returns the expected size for a set of source files and also verifies its correctness when calculating the size of source directories with nested subdirectories and files.

# Changes Made
- Added test functions to verify the behavior of the calculates_size function:
    - The first test checks if the function returns the expected size for a set of source files.
    - The second test verifies the correctness of the function when calculating the size of source directories with nested subdirectories and files.
- Added the calculates_size function to calculate the total size of the given source directories or file paths. It iterates through the provided list of sources and sums up their sizes. The result is displayed in a human-readable format.

# Motivation and Context
The addition of the calculates_size function enhances the functionality of the application by providing users with valuable information about the sizes of their source files and directories. The new tests ensure the proper functioning of the calculates_size function, improving the reliability and accuracy of the codebase.

# How to Test
1. Ensure that the project dependencies are installed.
2. Run the newly added test functions to verify the behavior of the calculates_size function.
3. Use different sets of source files, including both individual files and directories with nested subdirectories, to calculate their sizes using the calculates_size function. Verify that the returned sizes match the expected results.
4. Check that the sizes are displayed correctly in a human-readable format.
5. Ensure that the existing functionality of the application remains unaffected by the new additions.

# Related Issue(s)
N/A

# Checklist

Please review and merge this pull request at your earliest convenience. Any feedback or suggestions for improvement are welcome. Thank you!
